### ISSUE TITLE
[Day 4] CNN으로 CIFAR-10 분류 완료 - {박민호}

### ISSUE THREAD

**Issue by minho-03** [minho-03/algorithm#8]
## 실습 완료 내용

- [x] Kaggle Learn Lesson 1~3 완료 (The Convolutional Classifier, Convolution and ReLU, Maximum Pooling)

- [x] CIFAR-10 데이터 로드 & 탐색

- [x] CNN 모델 구축 (Conv2D + MaxPooling + Dense)

- [x] 모델 훈련 (조기 종료 적용)

- [x] 성능 평가 및 혼동행렬 분석

- [x] Dense Network와 성능 비교

## 데이터셋 정보

- 이미지 크기: 32×32 (컬러)

- 훈련 데이터: 40,000장

- 검증 데이터: 10,000장

- 테스트 데이터: 10,000장

- 클래스: 10가지 (airplane, automobile, bird, ..., truck)

## 모델 구축

- Conv2D(32, 3×3) + ReLU + MaxPooling(2×2)

- Conv2D(64, 3×3) + ReLU + MaxPooling(2×2)

- Conv2D(128, 3×3) + ReLU + MaxPooling(2×2)

- Flatten()

- Dense(256) + ReLU + Dropout(0.5)

- Dense(10) + Softmax (10-class 분류)

## 성능 결과

- 훈련 정확도: 91.50% 

- 검증 정확도: 76.17% 

- 테스트 정확도: 75.63% (최종)

- 총 파라미터: 620,362개

- 훈련 에포크: 11개 (조기 종료 적용)

## Dense Network와의 비교

| 항목 | Dense | CNN |

|------|-------|-----|

| 입력 형태 | 1D 벡터 | 2D 이미지 |

| 파라미터 수 | 약 77M | 620,362개 |

| 정확도 | 70~78% | 75.63% |

| 훈련 시간 | 2~5분 | 4분 |

| 메모리 | ~2GB | 2.37MB |

## 관찰 기록

1. CNN이 Dense Network보다 효율적인 이유:

   - 파라미터 감소: Dense 모델 대비 파라미터 수가 약 124배 적음에도 불구하고 비슷한 성능을 냄.

   - 공간 정보 보존: 이미지를 펴지 않고 2D 구조를 유지하여 사물의 형태를 더 잘 파악함.

   - 특징 학습의 계층성: CNN은 층이 깊어질수록 단순한 특징에서 복잡한 특징으로 단계별 학습

2. 가장 잘 분류되는 클래스 Top 3:

   - automobile(정확도: 89.40%)

   - ship(정확도: 80.70%)

   - airplane(정확도: 82.50%)

3. 가장 어려운 클래스 (정확도 낮음):

   - cat(정확도: 62.70%) → 이유: 개(dog)와 생김새가 비슷하고 특징이 복잡해서

   - dog(정확도: 63.70%) → 이유: 고양이(cat)와 혼동되는 경우가 많아서

4. 가장 많이 혼동되는 클래스 쌍:

   - cat→ dog(175건)

   - dog→ cat(141건)

## 개선 시도 (선택)

- 시도 1: (예: 필터 개수 증가) → 결과: ____%

- 시도 2: (예: Batch Normalization 추가) → 결과: ____%

- 가장 효과 있었던 변경: 

## 다음 단계 (내일)

Day 5: YOLO로 객체 검출 도전 예정

## 커밋 및 링크

- Kaggle 노트북: (URL)

- GitHub: (링크)

**Comment by minho-03** [comment]
<10가지 클래스 샘플 이미지>
<img width="408" height="799" alt="Image" src="https://github.com/user-attachments/assets/082a13bd-07ae-46cd-9970-0b679020b4a3" />
<모델 요약>
<img width="408" height="799" alt="Image" src="https://github.com/user-attachments/assets/605d603a-20c3-4f37-9e13-229b7b40ff79" />
<훈련 로그>
<img width="790" height="358" alt="Image" src="https://github.com/user-attachments/assets/16dfe255-ed15-4fcb-adcd-9257609da868" />
<손실/정확도 그래프 + 신뢰도 통계>
<img width="982" height="556" alt="Image" src="https://github.com/user-attachments/assets/70c22a9e-435c-449b-a9cf-7a4fcbaec543" />
<혼동행렬 (10×10) + 분류 리포트 + 저신뢰도 샘플>
<img width="625" height="758" alt="Image" src="https://github.com/user-attachments/assets/bb5d9d8b-c81f-45b2-94bd-62c43fcea4e8" />
